UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/NoticesToMariners/NoticesToMarinersControllersTest.cs

```diff
@@ -138,5 +139,32 @@ public void WhenCallDownloadWeeklyFileWithInvalidData_ThenReturnException()
                .And.Message.EqualTo("Response status code does not indicate success: 404 (Not Found).")
                , async delegate { await _nMController.DownloadWeeklyFile(batchId, filename, mimeType); });
         }
+
+        [Test]
+        public async Task WhenCallDownloadDailyFile_ThenReturnFile()
+        {
+            const string batchId = "44e8cce6-e69d-46bd-832d-6fd3d4ef8740";
+            const string filename = "SERIAL_D2022_18.txt";
+            const string mimeType = "application/text";
+
+            ActionResult result = await _nMController.DownloadDailyFile(batchId, filename, mimeType);
+            Assert.IsTrue(((FileContentResult)result) != null);
+            Assert.AreEqual("application/text",((FileContentResult)result).ContentType);            
+            Assert.AreEqual(1229033, ((FileContentResult)result).FileContents.Length);
+            Assert.AreEqual("https://filesqa.admiralty.co.uk", Config.BaseUrl);
+        }
+
+        [Test]
+      
+        public async Task WhenCallDownloadDailyFileWithInvalidData_ThenReturnException()
+        {
+            const string batchId = "08e8cce6-e69d-46bd-832d-6fd3d4ef8740";
+            const string filename = "Test.txt";
+            const string mimeType = "application/txt";
+
+            ActionResult result = await _nMController.DownloadDailyFile(batchId, filename, mimeType);
+            Assert.AreEqual(false,((RedirectToActionResult)result).PreserveMethod);
+            Assert.AreEqual("ShowDailyFiles", ((RedirectToActionResult)result).ActionName);
```

Comment:
This test is not returning any exception.

Reply:
Renamed WhenCallDownloadDailyFileWithInvalidData_ThenReturnException to WhenCallDownloadDailyFileWithInvalidData_ThenReturnShowDailyFiles